src/Graphing.hs

```diff
@@ -425,20 +444,24 @@ color graph extractSet update origin extractProperty nodesToColor = gmap applyCo
           coloredNodeSet = if (extractProperty node) `Set.member` nodesToColor then nodeSet `Set.union` Set.fromList [extractProperty origin] else nodeSet
       update coloredNodeSet node
 
--- | Gets all successors originating from 'a' on condition (excluding a)
+-- | Retrieve a node's children so long as it satisfies a specific condition.
+--   The list of reachable nodes excludes the origin.
+--
+--   Given the graph:
 --
--- Example
+-- >    1 -> 2
+-- >     \     \
+-- >      3     \
+-- >       \      4
+-- >         6     \
+-- >                5
 --
---   1 -> 2
---    \    \
---     3    \
---      \    4
---       6    \
---             5
+--   Here is an example to retrieve all reachable of nodes from 1, where the reachable nodes are not 2:
 --
--- reachableSuccessorsWithCondition [(1, 2), (1, 3), (2, 4), (3, 6), (4, 5)]  1  (\x -> x `Set.notMember conditionalSet) conditionalSet= Set.fromList [2] =
+-- >  reachableSuccessorsWithCondition [(1, 2), (1, 3), (2, 4), (3, 6), (4, 5)]  1  (\x -> x `Set.notMember conditionalSet) conditionalSet= Set.fromList [2]
+-- >  Returns: Set (3, 6)
 --
--- Set (3, 6)
+--   In the context of Maven submodule filtering (Maven/Common.hs), `reachableSuccessorsWithCondition` returns a submodule's dependencies given that the dependency is not a submodule.
```

Comment:
This description should be edited to explain what the function does without reference to Maven because `src/Graphing.hs` is mostly agnostic to what's actually in the graphs it manipulates. What you're really doing is saying that we ignore cycles when finding successors, correct?

Reply:
That's correct the work in this pr is to ignore cycles when finding successors.

I should've made note of this, but I am also reformatting the comments for the maven submodule functions and the graphing functions that I added. This was based on a previous discussion that I had with Jess bc it was mentioned that it was hard to distinguish the purpose of these functions and it's intended use

Reply:
ah ok, that's really reasonable. When I read this initially I thought it was suggesting that this was made especially rather than simply as an example. Thanks!